src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,101 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const emptyState = {
+    count: 0,
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+    title: '',
+    description: '',
+  };
+
+  const [field, setField] = useState(emptyState);
+
+  const unSuccessForm = !(
+    field.title
+    && field.imgUrl
+    && field.imdbUrl
+    && field.imdbId
+  );
+
+  const handleChange = (newValue: string, inputName: string) => {
+    setField({
+      ...field,
+      [inputName]: newValue,
+    });
+  };
+
+  const resetFields = () => {
+    setField(state => ({
+      ...emptyState,
+      count: state.count + 1,
+    }));
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onAdd({
+      title: field.title,
+      description: field.description,
+      imgUrl: field.imgUrl,
+      imdbUrl: field.imdbUrl,
+      imdbId: field.imdbId,
```

Comment:
You can destructure the field object before and shorten this 
```suggestion
      title,
      description,
      imgUrl,
      imdbUrl,
      imdbId,
```